### PR TITLE
SEDs should stay in active mode while a fail-safe is armed.

### LINKED ISSUE #24047: Staying in active mode through end of commissioning fails if commissioning window closes before commissioning ends
Even after https://github.com/project-chip/connectedhomeip/pull/23936 we can end up going back to sleep in the middle of commissioning as follows:

1. A commissioning window is opened for 5 mins
2. We enter active mode.
3. Foud (4) minutes later a PASE session is established and commissioning starts.
4. One (1) minute after that the commissioning window timer fires and we close the commissioning window, which takes us out of active mode, even though commissioning may not be done.

What we should probably do is that if the commissioning window is closed while we have a live PASE session we stay in active mode until the PASE session is closed.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Fixes https://github.com/project-chip/connectedhomeip/issues/24047

Implements spec fix https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/5683

### CODE REVIEW

File: src/app/FailSafeContext.h [hunk 1]
```diff
@@ -82,7 +82,7 @@ class FailSafeContext
 
     bool MatchesFabricIndex(FabricIndex accessingFabricIndex) const
     {
-        VerifyOrDie(mFailSafeArmed);
+        VerifyOrDie(IsFailSafeArmed());
```
Comment (cecille) [inline src/app/FailSafeContext.h:85]: Doesn't this cause a recursive problem where is IsFailsafeArmed calls this function? Or are there two versions of that function?

File: src/app/FailSafeContext.h [hunk 2]
```diff
@@ -66,7 +66,7 @@ class FailSafeContext
 
     bool IsFailSafeArmed(FabricIndex accessingFabricIndex) const
     {
-        return mFailSafeArmed && MatchesFabricIndex(accessingFabricIndex);
+        return IsFailSafeArmed() && MatchesFabricIndex(accessingFabricIndex);
```
Comment (cecille) [inline src/app/FailSafeContext.h:69]: As below, is this a recursive call?

**Review by cecille — APPROVED** [review]

File: src/app/FailSafeContext.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/FailSafeContext.h:85]: There are two IsFailSafeArmed(): one takes no args and checks whether the failsafe is armed at all, and one that takes a fabric index and checks whether the failsafe is armed for that specific fabric index.

This is called from the version that takes a fabric index, and it's calling the one that does not take a fabric index.

File: src/app/FailSafeContext.h (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/FailSafeContext.h:69]: It's not; there are two different overloads of IsFailSafeArmed.

**Review by andy31415 — APPROVED** [review]

**Review by tcarmelveilleux — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**